android/src/main/java/com/tailscale/ipn/App.java

```diff
@@ -94,6 +103,25 @@ String getHostname() {
 		return manu + " " + model;
 	}
 
+	// get user defined nickname from Settings
+	private String getUserConfiguredDeviceName() {
+		String nameFromSystemBluetooth = Settings.System.getString(getContentResolver(), "bluetooth_name");
+		String nameFromSecureBluetooth = Settings.Secure.getString(getContentResolver(), "bluetooth_name");
+		String nameFromSystemDevice = Settings.Secure.getString(getContentResolver(), "device_name");
+
+		Log.d("com.tailscale.ipn.App", "Device name from System Bluetooth: " + nameFromSystemBluetooth);
+		Log.d("com.tailscale.ipn.App", "Device name from Secure Bluetooth: " + nameFromSecureBluetooth);
+		Log.d("com.tailscale.ipn.App", "Device name from System Device: " + nameFromSystemDevice);
+
+		if (notEmpty(nameFromSystemBluetooth)) return nameFromSystemBluetooth;
+		if (notEmpty(nameFromSecureBluetooth)) return nameFromSecureBluetooth;
+		return nameFromSystemDevice;
+	}
+
+	private static boolean notEmpty(String str) {
```

Comment:
negatively named boolean functions inevitably get double negatived and read confusingly later.

better to just name it `emptyStr` and then write `!emptyStr` at the call sites.

Reply:
absolutely right. my bad.